controllers/component/component_adapter.go

```diff
@@ -52,6 +55,17 @@ func NewAdapter(component *applicationapiv1alpha1.Component, application *applic
 	}
 }
 
+// EnsureComponentHasFinalizer is an operation that will ensure components
+// that have been created are assigned a finalizer
+func (a *Adapter) EnsureComponentHasFinalizer() (controller.OperationResult, error) {
```

Comment:
Maybe we should check if the component is not being deleted (with something like `if hasComponentBeenDeleted(a.component)`) so we don't add a finalizer to such component unnecessarily in that case.

Reply:
good idea, added that [here](https://github.com/redhat-appstudio/integration-service/pull/410/files#diff-0a268eab99b22ffc9ca9a74ef2752ed7dd05a822b3da8fe160ffee3a2ad1035eR61)

Reply:
Hm, looking at it now, I think the check should be inverse (`! hasComponentBeenDeleted(a.component)`) because currently it only adds the finalizer to components that begin to be deleted.

Reply:
Good catch, thanks - fixed in most recent prs.

Reply:
Yup, thanks!